src/plugins/index_pattern_editor/public/components/empty_prompts/empty_prompts.tsx

```diff
@@ -47,31 +47,30 @@ export const EmptyPrompts: FC<Props> = ({ allSources, onCancel, children, loadSo
   } = useKibana<IndexPatternEditorContext>();
 
   const [remoteClustersExist, setRemoteClustersExist] = useState<boolean>(false);
+  const [hasCheckedRemoteClusters, setHasCheckedRemoteClusters] = useState<boolean>(false);
+
   const [goToForm, setGoToForm] = useState<boolean>(false);
 
   const hasDataIndices = allSources.some(isUserDataIndex);
   const hasUserIndexPattern = useAsync(() =>
     indexPatternService.hasUserIndexPattern().catch(() => true)
   );
 
-  useCallback(() => {
-    let isMounted = true;
-    if (!hasDataIndices)
+  useEffect(() => {
```

Comment:
This is the main fix `[DataViews] Fix checking remote clusters in empty state`

We had `useCallback` instead of `useEffect` by mistake